app/src/main/java/org/p2p/wallet/deeplinks/DeeplinkData.kt

```diff
@@ -0,0 +1,19 @@
+package org.p2p.wallet.deeplinks
+
+import android.content.Intent
+
+/**
+ * Container for deeplink information, such as target screen, extra paths and arguments
+ * @param target @see deeplink target
+ * @param pathSegments [List] of string (the same as in Uri)
+ * @param args extra arguments
+ * @param intent original intent, useful for getting extras from notification
+ */
+data class DeeplinkData(
+    val target: DeeplinkTarget,
+    val pathSegments: List<String> = emptyList(),
+    val args: Map<String, String?> = emptyMap(),
+    val intent: Intent? = null
+) {
+    fun hasArgNotNull(key: String): Boolean = args[key] != null
```

Comment:
`hasArg` is ok too
Map<String, **String?**> is needed? 
args[key] will return null if a key is not found in the map 
Have you tried just Map<String, String>?

Reply:
- the value can be omitted when passed to the application, so we can receive null values, that's why the value is nullable.
- `hasArgNotNull`: this is more explicit because otherwise it's unclear why we shouldn't just check using `args.containsKey(key)`, but the function's name is still discussable
- using an empty map is preferable over a nullable one, as we can check values without verifying if the map is null.

Reply:
- if it's omitted, then there will be no key inside the map? or we will still have something like {"key": null}